tests/data/specs/checksum_message.rflx

```diff
@@ -0,0 +1,17 @@
+package Checksum_Message is
+
+   type Field_A is mod 2**8;
+   type Field_B is mod 2**8;
+   type Checksum is mod 2**8;
+
+   type Message is
+      message
+         Field_A : Field_A;
+         Field_B : Field_B;
+         Checksum : Checksum
+            then null
+               if Checksum'Valid_Checksum;
+      end message with
+         Checksum => (Checksum => (Field_A'First .. Field_A'Last, Field_B'First .. Field_B'Last));
```

Comment:
The checksum would be identical for all possible instances of `Message`, as the bounds of all fields are static. Is that intended? Wouldn't it make sense to use a more realistic checksum here? That would at least enable the reuse of this specification for other checksum-related tests.

Couldn't an existing specification be reused for the regression test, e.g., `TLV_With_Checksum`? Is there any advantage in adding this spec?

Reply:
>Wouldn't it make sense to use a more realistic checksum here? 

I don't think that it is necessary to use a realistic checksum function as the test is only about checking, if message is invalid when the parsed checksum value is not the same as the value returned by the checksum function. There are other tests that check the correct calculation of the checksum. There is also already a test `test_checksum_invalid` (which uses icmp and the icmp checksum function) that was supposed to test this in the first place, but the bug was not triggered by it.

>Couldn't an existing specification be reused for the regression test, e.g., TLV_With_Checksum?

Changed it. I used the other spec originally because I was sure the bug is triggered by it and it is the same spec that is used in the validation tool tests.

Reply:
> There is also already a test test_checksum_invalid (which uses icmp and the icmp checksum function) that was supposed to test this in the first place, but the bug was not triggered by it.

What is the significant difference which triggered the bug? It would be good to make the difference clear in the name of the test procedure instead of just adding `_tlv`.

Reply:
In `ICMP`, the checksum field is in the middle of the message, but the checksum can only be calculated when all fields have been set. When parsing the message, the checksum was not recalculated when the checksum field had already been set or if the checksum function had been executed before. That was introduced because when setting fields manually, the checksum needs to be recalculated if an already set field changes. 

In `TLV_With_Checksum`, the checksum field is the last field of the message and all other fields (except for Checksum) are needed for the checksum calculation. As soon as the Data field had been parsed, the automatic checksum calculation was triggered because the checksum field had not been set. Then the checksum field was parsed and the before set value overridden. But after that, because the checksum function had been executed before, another automatic checksum calculation was triggered and the parsed value overridden. 

The main difference is not the location of the checksum field, but rather if the checksum field was parsed before the automatic checksum calculation was triggered for the first time. Idk how to express that in a few words so that it makes sense for a reader in a few weeks... and there is no automatic checksum calculation anymore when parsing a message.

Reply:
Let's leave it as it is.